Core/Utilities/RabbitMQ.py

```diff
@@ -0,0 +1,144 @@
+"""
+Class for management of RabbitMQ connections
+"""
+
+__RCSID__ = "$Id$"
+
+import datetime
+import json
+import threading
+import stomp
+from DIRAC import gLogger, gConfig, S_OK, S_ERROR
+
+class RabbitInterface( object ):
+  """
+  Class for management of RabbitMQ connections
+  Allows to both send and receive messages from a queue
+  The class also implements callback functions to be able to act as a listener and receive messages
+  """
+
+  msgList = []
+  lock = threading.Lock()
+
+  def on_error( self, headers, message ):
+    """
+    Callback function called when an error happens
+    """
+    gLogger.error( message )
+
+  def defaultCallback( self, headers, message ):
+    """
+    Default callback function called every time something is read from the queue
+    """
+    dictionary = json.loads( message )
+    with self.lock:
+      self.msgList.append( dictionary )
+
+  def setupConnection( self, system, queueName, receive = False, messageCallback = None ):
+    """
+    Establishes a new connection to RabbitMQ
+    system indicates in which System the queue works
+    queueName is the name of the queue to read from/write to
+    receive indicates whether this object will read from the queue or read from it
+    messageCallback is the function to be called when a new message is received from the queue (only receiver mode).
+    If None, the defaultCallback method is used instead
+    """
+    self.receiver = receive
+
+    if self.receiver:
+      if messageCallback:
+        self.on_message = messageCallback
+      else:
+        self.on_message = self.defaultCallback
+
+    self.queueName = queueName
+    setup = gConfig.getValue( '/DIRAC/Setup', '' )
+
+    result = gConfig.getOption( '/Systems/%s/%s/MessageQueueing/%s/Host' % ( system, setup, queueName ) )
```

Comment:
All this code up to line 70 seems the same mabye one could refactor by adding some function that would take one string as argument "host", "port" etc